src/main/java/com/fasterxml/jackson/core/sym/ByteQuadsCanonicalizer.java

```diff
@@ -961,16 +990,16 @@ public String addName(String name, int[] q, int qlen)
         return name;
     }
 
-    private void _verifySharing()
+    private void _verifySharing() throws StreamConstraintsException
     {
         if (_hashShared) {
             // 12-Mar-2021, tatu: prevent modifying of "placeholder" and
             //   parent tables
             if (_parent == null) {
                 if (_count == 0) { // root
-                    throw new IllegalStateException("Cannot add names to Root symbol table");
+                    throw new StreamConstraintsException("Cannot add names to Root symbol table");
```

Comment:
Ok, I'll change this back, actually, since it's internal sanity check (not something users can trigger). Same for next one.